include/deal.II/base/partitioner.templates.h

```diff
@@ -403,11 +403,37 @@ namespace Utilities
           // local values. For insert, nothing is done here (but in debug mode
           // we assert that the specified value is either zero or matches with
           // the ones already present
-          if (vector_operation != dealii::VectorOperation::insert)
+          if (vector_operation == dealii::VectorOperation::add)
             for (; my_imports != import_indices_data.end(); ++my_imports)
               for (unsigned int j = my_imports->first; j < my_imports->second;
                    j++)
                 locally_owned_array[j] += *read_position++;
+          else if (vector_operation == dealii::VectorOperation::min)
+            for (; my_imports != import_indices_data.end(); ++my_imports)
+              for (unsigned int j = my_imports->first; j < my_imports->second;
+                   j++)
+                {
+                  // To ensure that this code also compiles with
+                  // complex numbers, we only compare the real part of
+                  // the variable. Note that min/max do not make sense
+                  // with complex numbers.
+                  if (std::real(*read_position) -
+                        std::real(locally_owned_array[j]) <
+                      0.0)
```

Comment:
This decision is something we could discuss: The problem is that we use complex numbers, so we cannot simply use `std::min(local_array[j], *read_position)` here. I think the solution @schoeder proposes makes sense as it restricts the check to the real part (even though it might take a bit longer to execute). Or does anybody want to see another solution? Say a different compare function for real and complex valued arguments?

Reply:
What if you just disallowed complex-valued vectors here? `min` and `max` are just not defined for complex numbers.